src/safeposix/syscalls/fs_calls.rs

```
@@ -785,43 +846,63 @@ impl Cage {
                         socket_inode_obj.linkcount += 1; //add link to inode
                     }
 
+                    // Directory type inode is not supported for linking, so return an error.
                     Inode::Dir(_) => {
```

can we reorder this to fail early

---

Done